### PR TITLE
2221 Remove placeholder on editing outbound shipment stock

### LINKED ISSUE #2221: Outbound Shipment: editing placeholder line & allocating stock doesn't delete the placeholder
## What went wrong? 😲
As per title

## Expected behaviour 🤔
Placeholders should disappear if they have zero quantity

## How to Reproduce 🔨

1. Go to outbound shipments
2. Add items from master list
3. Edit item that has stock, allocate quantity
4. Click OK > back in the list the placeholder still exists with zero quantity

CleanShot 2023-09-12 at 17 31 35@2x

## Your environment 🌱

   - [X] browser - Firefox Mac
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by raviSussol**
Fixes #2221 

# 👩🏻‍💻 What does this PR do? 
 
Fix the logic for getting `deleteOutboundShipmentUnallocatedLines` array to delete placeholder while saving outbound shipment line with stock.

# 🧪 How has/should this change been tested? 

1. Go to outbound shipments
2. Add items from master list
3. Edit item that has stock, allocate quantity
4. Click OK > back in the list the placeholder will be replaced by the line with stock filled in

### CODE REVIEW

**Review by roxy-dao — APPROVED** [review]
Thanks Ravi~

File: client/packages/invoices/src/OutboundShipment/api/api.ts [hunk 1]
```diff
@@ -374,7 +374,7 @@ export const getOutboundQueries = (sdk: Sdk, storeId: string) => ({
           ({ type, numberOfPacks, isUpdated, isCreated }) =>
             type === InvoiceLineNodeType.UnallocatedStock &&
             numberOfPacks === 0 &&
-            isUpdated &&
+            !isUpdated &&
```
Comment (mark-prins) [inline client/packages/invoices/src/OutboundShipment/api/api.ts:377]: can you just remove the `isUpdated` check please. If a line has a placeholder quantity, and then stock comes in and the line is allocated, then the placeholder quantity could be reduced to `0` and then the `isUpdated` flag is set to true. We'd want those placeholders removed too!

**Review by mark-prins — CHANGES_REQUESTED** [review]
Thanks @raviSussol - that's it. I think just remove the updated flag check though - if the placeholder is `0` then we should just remove it

File: client/packages/invoices/src/OutboundShipment/api/api.ts (hunk 1 above)
Comment (raviSussol) [reply client/packages/invoices/src/OutboundShipment/api/api.ts]: Ah yes. I'd been wondering when `isUpdated` flag is set to true cause I hadn't tested the case you've mentioned for the placeholder. Now I know. Cool. 27ccee7b7

**Review by mark-prins — APPROVED** [review]
👍 sweet - thanks for the change too!

**Pull request merged**